src/Page.js

```diff
@@ -566,43 +566,32 @@ class Page {
    * Produces expressive layouts by inserting page data into pre-specified layout
    * @param pageData a page with its front matter collected
    * @param {FileConfig} fileConfig
+   * @param {Parser} markbinder instance from the caller, for adding the seen sources.
    */
-  generateExpressiveLayout(pageData, fileConfig) {
-    const markbinder = new MarkBind();
-    const template = {};
-    template[LAYOUT_PAGE_BODY_VARIABLE] = pageData;
+  generateExpressiveLayout(pageData, fileConfig, markbinder) {
```

Comment:
passing the parser (`markbinder`) isn't very clean, but will be dealt with in #1189 in full (making it an instance variable)